### PR TITLE
refactor: Remove color ramps from UI (with backwards-compatibility)

### PR DESCRIPTION
**Pull request by ShrimpCryptid**
Problem
=======
First part of #217. Adds the ability to hide color ramps and palettes from the UI while still allowing them to be loaded via the URL. This supports backwards-compatibility with old or outdated ramps.

*Estimated review time: small, 15 minutes* 

Solution
========
- Refactors the color ramps and categorical palettes to have a list of known data and a separate list of displayed keys.
- Refactors `ColorRampDropdown` to only show palettes in the list of displayed keys.
- Updates unit tests

## Type of change
* New feature (non-breaking change which adds functionality)

Steps to Verify:
----------------
1. Open the preview link and observe that a color map was loaded from the URL that does not appear in the list of options: https://allen-cell-animated.github.io/nucmorph-colorizer/pr-preview/pr-245/?dataset=https%3A%2F%2Fraw.githubusercontent.com%2Fallen-cell-animated%2Fnucmorph-colorizer%2Fmain%2Fdocumentation%2Fexample_dataset%2Fmanifest.json&feature=Continuous%20Feature&color=esri-orange_5&palette-key=iwanthue_pastel_3
3. Switch to a categorical feature. The displayed color palette should not match any in the dropdown.

Screenshots (optional):
-----------------------
*Notice that "ESRI Orange 5" is not in the list of options but is still being rendered!*

image

### CODE REVIEW

File: src/components/Dropdowns/ColorRampDropdown.tsx [hunk 1]
```diff
@@ -20,10 +20,16 @@ import styles from "./ColorRampDropdown.module.css";
 type ColorRampSelectorProps = {
   selectedRamp: string;
   onChangeRamp: (colorRampKey: string, reversed: boolean) => void;
-  colorRamps?: Map<string, ColorRampData>;
+  /**
+   * The keys of the color ramps to display, in order. If `null` (default), shows all
+   * the color ramps given by `knownColorRamps`.
+   */
+  colorRampsToDisplay?: string[] | null;
+  knownColorRamps?: Map<string, ColorRampData>;
```
Comment (interim17) [inline src/components/Dropdowns/ColorRampDropdown.tsx:28]: If the purpose of this feature is to hide these ramps, but make them accessible by URL, why build in this fall back?

File: src/components/Dropdowns/ColorRampDropdown.tsx [hunk 2]
```diff
@@ -61,6 +69,10 @@ const ColorRampSelector: React.FC<ColorRampSelectorProps> = (propsInput): ReactE
   const props = { ...defaultProps, ...propsInput } as Required<ColorRampSelectorProps>;
   const theme = useContext(AppThemeContext);
 
+  const colorRampsToDisplay = props.colorRampsToDisplay ?? Array.from(props.knownColorRamps.keys());
+  const categoricalPalettesToDisplay =
+    props.categoricalPalettesToDisplay ?? Array.from(props.knownCategoricalPalettes.keys());
+
```
Comment (interim17) [inline src/components/Dropdowns/ColorRampDropdown.tsx:75]: Same question as above, if the known ramps and palettes are truly legacy and meant to be accessed only by URL, why provide them as a fallback?

**Review by interim17 — APPROVED** [review]
Non-blocking questions left, LGTM.

File: src/components/Dropdowns/ColorRampDropdown.tsx (hunk 1 above)
Comment (ShrimpCryptid) [reply src/components/Dropdowns/ColorRampDropdown.tsx:28]: It was originally in for ease of testing. I've removed it!

File: src/components/Dropdowns/ColorRampDropdown.tsx [hunk 3]
```diff
@@ -20,10 +20,15 @@ import styles from "./ColorRampDropdown.module.css";
 type ColorRampSelectorProps = {
   selectedRamp: string;
   onChangeRamp: (colorRampKey: string, reversed: boolean) => void;
-  colorRamps?: Map<string, ColorRampData>;
+  /**
+   * The keys of the color ramps to display, in order.
+   */
```
Comment (frasercl) [inline src/components/Dropdowns/ColorRampDropdown.tsx:25]: optional micro-nit: could be single-line doc comment

Looking further down the file, I see some short JSDocs are expanded like this one, and some are single-line. Is there a pattern to which gets which? e.g. I see most items with multiline comment blocks (except this one) are at the top level.

again this is so tiny you're absolutely welcome to ignore it

File: src/components/Dropdowns/ColorRampDropdown.tsx [hunk 4]
```diff
@@ -61,6 +66,9 @@ const ColorRampSelector: React.FC<ColorRampSelectorProps> = (propsInput): ReactE
   const props = { ...defaultProps, ...propsInput } as Required<ColorRampSelectorProps>;
   const theme = useContext(AppThemeContext);
 
+  const colorRampsToDisplay = props.colorRampsToDisplay;
+  const categoricalPalettesToDisplay = props.categoricalPalettesToDisplay;
```
Comment (frasercl) [inline src/components/Dropdowns/ColorRampDropdown.tsx:70]: another tiny annoying suggestion: could be one destructuring assignment, like so: `const { colorRampsToDisplay, categoricalPalettesToDisplay } = props;`

**Review by frasercl — APPROVED** [review]
Looks great! Love the consideration for backwards compatibility.

File: src/components/Dropdowns/ColorRampDropdown.tsx (hunk 3 above)
Comment (ShrimpCryptid) [reply src/components/Dropdowns/ColorRampDropdown.tsx:25]: Fixed!

**Pull request merged**